### PR TITLE
feat: lambda-promtail; ensure messages to Kinesis are usable by refactoring parsing of KinesisEvent to match parsing of CWEvents + code cleanup

### LINKED ISSUE #10544: Review labels consistency in lambda-promtail
Lambda promtail adds the `__aws_log_type` label to entries, to tell apart which was the logging source of that entry. This label is added for s3-based logs, but not for cloudwatch logs for example. 

We should review the supported sources, and make these labels consistent.

### PR DESCRIPTION
**Pull request by HatiCode**
Refactor parsing of KinesisEvent to match parsing of CWEvents + code cleanup

**What this PR does / why we need it**:
This PR refactors the kinesis.go parsing code to match the way CWEvents are sent to Promtail.
When Kinesis is chosen as a delivery method it should still have a simple and usable log format. So far the message in the DATA_MESSAGE is unusable. This is solved by assigning labels to the logEvents in the DATA_MESSAGE the same way it's done in cw.go

**Which issue(s) this PR fixes**:
Fixes #10544

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by cstyan** [comment]
hey @HatiCode there are some lambda promtail tests failing

**Comment by HatiCode** [comment]
> hey @HatiCode there are some lambda promtail tests failing

Fixed.

File: tools/lambda-promtail/lambda-promtail/kinesis.go [hunk 1]
```diff
@@ -79,3 +79,39 @@ func ungzipData(data []byte) ([]byte, error) {
...
+	labels := model.LabelSet{
+		model.LabelName("__aws_log_type"):                 model.LabelValue("kinesis"),
+		model.LabelName("__aws_kinesis_event_source_arn"): model.LabelValue(record.EventSourceArn),
+		model.LabelName("__aws_cloudwatch_log_group"):     model.LabelValue(recordData.LogGroup),
+		model.LabelName("__aws_cloudwatch_owner"):         model.LabelValue(recordData.Owner),
+	}
+
+	if keepStream {
+		labels[model.LabelName("__aws_cloudwatch_log_stream")] = model.LabelValue(recordData.LogStream)
+	}
```
Comment (cstyan) [inline tools/lambda-promtail/lambda-promtail/kinesis.go:99]: should these not be prefixed with `__aws_kinesis` instead of `__aws_cloudwatch`?

Comment (HatiCode) [reply tools/lambda-promtail/lambda-promtail/kinesis.go:99]: @cstyan depending on the pov. The values are coming from cloudwatch, not from the K stream. K is a data stream not the source so I believe you want to have a filter that's a reflection of what your actual cloudwatch looks like, not the tool you used in the middle to stream that data.

Comment (cstyan) [reply tools/lambda-promtail/lambda-promtail/kinesis.go:99]: that makes sense, thanks for the context :+1:

**Review by cstyan — APPROVED** [review]

**Pull request merged**